io/zenoh-transport/tests/unicast_openclose.rs

```diff
@@ -680,3 +680,153 @@ tOzot3pwe+3SJtpk90xAQrABEO0Zh2unrC8i83ySfg==
 
     task::block_on(openclose_transport(&endpoint));
 }
+
+#[cfg(all(feature = "transport_serial", target_os = "linux"))]
+#[test]
+fn openclose_serial_only() {
+    task::block_on(async {
+        zasync_executor_init!();
+    });
+
+    env_logger::try_init().unwrap_or_else(|_| log::warn!("`env_logger` already initialized"));
+
+    async fn create_ports() {
+        let args = [
+            "PTY,link=ttyVIRT0".to_string(),
+            "PTY,link=ttyVIRT1".to_string(),
+        ];
+        log::trace!("starting process: socat {} {}", args[0], args[1]);
+
+        let process = std::process::Command::new("socat")
+            .args(&args)
+            .spawn()
+            .expect("unable to spawn socat process");
+        log::trace!(".... done! (pid: {:?})", process.id());
+    }
+
+    task::spawn(create_ports());
+
+    task::block_on(async move {
+        let endpoint_router: EndPoint = "serial/ttyVIRT0".parse().unwrap();
+        let endpoint_client: EndPoint = "serial/ttyVIRT1".parse().unwrap();
+        async_std::task::sleep(std::time::Duration::from_secs(10)).await;
+        openclose_serial_transport(&endpoint_router, &endpoint_client).await;
+    });
+}
+
+#[cfg(all(feature = "transport_serial", target_os = "linux"))]
+async fn openclose_serial_transport(router_endpoint: &EndPoint, client_endpoint: &EndPoint) {
```

Comment:
Instead of replicating the test, I suggest to adapt the arguments passed to the `openclose_transport` function.
E.g., an enum could be used to pass 1 or 2 endpoints and react accordingly. What do you think?